src/modules/app/components/routes.jsx

```diff
@@ -1,94 +1,72 @@
-import React, { Component } from 'react';
-import PropTypes from 'prop-types';
-import { ACCOUNT, CREATE_MARKET, TRANSACTIONS, M, MARKETS, MY_POSITIONS, MY_MARKETS, MY_REPORTS, AUTHENTICATION } from 'modules/app/constants/views';
-import { shouldComponentUpdateOnStateChangeOnly } from 'utils/should-component-update-pure';
+import React from 'react';
+import { Route, Switch, Redirect } from 'react-router-dom';
 
-// NOTE --  the respective routes are imported within the switch statement so that
-//          webpack can properly code split the views into independently loadable chunks
-export default class Routes extends Component {
-  static propTypes = {
-    activeView: PropTypes.oneOfType([
-      PropTypes.number,
-      PropTypes.string
-    ]).isRequired,
-    setSidebarAllowed: PropTypes.func.isRequired
-  }
+import App from 'modules/app/container';
 
-  constructor(props) {
-    super(props);
+import asyncComponent from 'modules/common/components/async-component';
```

Comment:
The purpose of `asyncComponent` appears to be to abstract away boilerplate present in the old `routes.jsx`.  While it does make the callsites in this file a bit cleaner, I think on balance introducing a new component class for this is overkill / is a net complexity increase.